lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,45 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        # Defer to IPython to handle html output if possible
+        if 'IPython' in sys.modules:
+            import IPython
+            ip = IPython.get_ipython()
+            # Check whether %matplotlib was run. Is there a better way?
+            ib_list = [c for c in ip.configurables 
+                       if 'InlineBackend' in type(c).__name__]
+            if ib_list:
+                return
```

Comment:
Just dived into IPython itself. It looks like we are also storing the sate on a function attribute (!!!).

```
In [1]: from IPython.core.pylabtools import configure_inline_support

In [2]: configure_inline_support.current_backend
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-eec19e858e54> in <module>
----> 1 configure_inline_support.current_backend

AttributeError: 'function' object has no attribute 'current_backend'

In [3]: %matplotlib qt

In [4]: configure_inline_support.current_backend
Out[4]: 'other'

In [5]: %matplotlib inline

In [6]: configure_inline_support.current_backend
Out[6]: 'inline'
```

Not sure it is better though.